pinot-spi/src/main/java/org/apache/pinot/spi/config/table/DedupConfig.java

```diff
@@ -20,25 +20,34 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
 import org.apache.pinot.spi.config.BaseJsonConfig;
 
 public class DedupConfig extends BaseJsonConfig {
+  @JsonPropertyDescription("Whether dedup is enabled or not.")
   private final boolean _dedupEnabled;
+  @JsonPropertyDescription("Function to hash the primary key.")
   private final HashFunction _hashFunction;
+  @JsonPropertyDescription("Custom class for dedup metadata manager.")
   private final String _metadataManagerClass;
+  @JsonPropertyDescription("Whether to use TTL for dedup metadata cleanup, it uses the same unit as the time column.")
+  private final double _metadataTTL;
```

Comment:
Just curious, would it be more readable if we use a string for the TTL (e.g., '1 day') or introduce two fields: one for the time value and one for the time unit?

Reply:
similar to upsert metadata manager, we use the same unit as the time column so it's easier to do the calculation - we don't need to covert them to the same unit